### ISSUE TITLE
Identify after_hook trigger

### ISSUE THREAD

**Issue by emilkor1** [gruntwork-io/terragrunt#3211]
### Summary

A way to identity the after_hook trigger if hook can be triggered by multiple commands

### Motivation

I am running an after_hook which executes on both commands plan and apply.

The hook starts a Go script which creates database users.

I wish to know within the script which of those has initiated the trigger such that I can print the outcome of what will happen when an apply will be performed (simply a dry run of the script).

### Proposal

A simplified version of the hook:

```
terraform {
  after_hook "create_users" {
    commands = ["plan", "apply"]
    execute = [
      "go", "run", "main.go",
    ]
  }
}
```

For my use case, this is a Go CLI with quite many arguments, thus I do not want to split it into two distinct after_hooks. I also have to update the CLI arguments on both hooks for it to work in that case.

My proposal is for example to add some type of environment variable such that I can do a os.Getenv(...) in the Go script.

### Technical Details

A non-breaking feature for hooks.

### Press Release

A new environment variable for hooks to distinguish between triggers.

### Drawbacks

No drawbacks.

### Alternatives

*No response*

### Migration Strategy

*No response*

### Unresolved Questions

Any alternative or existing solution?

### References

*No response*

### Proof of Concept Pull Request

*No response*

### Support Level

- [ ] I have Terragrunt Enterprise Support
- [ ] I am a paying Gruntwork customer

### Customer Name

*No response*

**Comment by levkohimins** [comment]
Hey @emilkor1, If ​​I understand your request correctly, you want the running application to know which command ("plan",  "apply", ... ) is running it, i.e. pass an env variable to the application, like `TERRAFORM_COMMAND` ?

**Comment by emilkor1** [comment]
Hey @levkohimins.

Yes, currently I have to send it in as a parameter and duplicate the code (as of my knowledge).

It is very tedious and does not really work so I only stick with the apply without really knowing if it is applying what I expect.

```hcl
terraform {
  source = "git::https://github.com/.../v1.0.0"

  after_hook "create_users" {
    commands = ["plan"]
    execute = [
      "go", "run", "-C", "main.go",
      "-host", "...",
      "-identity", "...",
      "-role", "user1",
      "-role", "user2",
      "-role", "user3",
      ...,
      "-cmd", "plan",      
    ]
  }

  after_hook "create_users" {
      commands = ["apply"]
      execute = [
        "go", "run", "-C", "main.go",
        "-host", "...",
        "-identity", "...",
        "-role", "user1",
        "-role", "user2",
        "-role", "user3",
        ...,
        "-cmd", "apply",      
      ]
    }
}
```

I wish to do the following:

```hcl
terraform {
  source = "git::https://github.com/.../v1.0.0"

  after_hook "create_users" {
    commands = ["plan", "apply"]
    execute = [
      "go", "run", "main.go",
      "-host", "...",
      "-identity", "...",
      "-role", "user1",
      "-role", "user2",
      "-role", "user3",
      ...,   
    ]
  }
}
```

If we get something like an `TERRAFORM_COMMAND` environment variable, I can perform a plan / apply programatically.

**Comment by levkohimins** [comment]
Resolved in [v0.61.1](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.61.1) release.
Make sure to read https://terragrunt.gruntwork.io/docs/features/hooks/
